pkg/patterns/declarative/reconciler.go

```diff
@@ -63,29 +69,44 @@ type DeclarativeObject interface {
 // For mocking
 var kubectl = kubectlcmd.New()
 
-func (r *Reconciler) Init(mgr manager.Manager, prototype DeclarativeObject, opts ...reconcilerOption) error {
+func (r *Reconciler) Init(mgr manager.Manager, prototype DeclarativeObject, opts ...reconcilerOption) (err error) {
```

Comment:
super nit: can you drop the named return here and in setReconcileMetrics? it makes it a bit harder to read the function once it's more than few lines.

Reply:
OK. I'll drop named return in `*Reconciler.Init` here. However, in `setReconcileMetrics`, we cannot use short variable declaration [here](https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/pull/98/files#diff-23c5c12ad7f72be6835d4f216e702363R370) because `gvks` will be redeclared & be in scope of 'if' block.

Should I use function scope variable `err`?

Reply:
In the remaining cases in setReconcileMetrics you should `return nil`, this makes it clear when reading the code that no error can be returned from this valid path.

Reply:
Thanks. I understand.